scripts/core/editor3/reducers/editor3.tsx

```diff
@@ -283,11 +289,17 @@ const onTab = (state, e) => {
         'ordered-list-item',
     ].includes(blockType)) { // let draft-js handle the Tab event
         newState = RichUtils.onTab(e, editorState, 4);
-    } else if (!e.shiftKey) {
-        const tabCharacter = '\t';
-        const newContent = Modifier.replaceText(editorState.getCurrentContent(), selection, tabCharacter);
-
-        newState = EditorState.push(editorState, newContent, 'insert-characters');
+    } else {
+        const tabOption = editorFormat.includes('tab');
+        const spacesOption = editorFormat.includes('tab as spaces');
+
+        if (tabOption && !e.shiftKey) {
+            newState = insertTab(editorState, selection, '\t');
+            e.preventDefault();
+        } else if (spacesOption && e.shiftKey) {
+            newState = insertTab(editorState, selection, '        ');
```

Comment:
looks like you can insert anything using that function, not only tabs. How about calling it `insertText` instead?